src/v/datalake/data_writer_interface.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "datalake/schemaless_translator.h"
+#include "iceberg/datatypes.h"
+#include "iceberg/values.h"
+
+#include <cstddef>
+#include <memory>
+
+#pragma once
+
+namespace datalake {
+class data_writer_result {
+public:
+    data_writer_result() = default;
+    data_writer_result(const data_writer_result&) = default;
+    data_writer_result(data_writer_result&&) = delete;
```

Comment:
generally i'd expect us to have deleted copy constructors instead of move constructors. what's the motivation?